src/plugins/auto/src/plugin.cpp

```diff
@@ -575,91 +575,72 @@ std::list<DeviceInformation> Plugin::get_valid_device(
     if (meta_devices.empty()) {
         OPENVINO_THROW("No available device to select in ", get_device_name());
     }
-
+    bool is_default_list = true;
+    std::list<DeviceInformation> valid_devices;
     std::list<DeviceInformation> CPU;
     std::list<DeviceInformation> dGPU;
     std::list<DeviceInformation> iGPU;
-    std::list<DeviceInformation> MYRIAD;
-    std::list<DeviceInformation> VPUX;
+    std::list<DeviceInformation> Others;
+    auto is_supported_model = [this, model_precision](const std::string& device_name) {
+        // Check if candidate device supported the specified model precision.
+        auto capability = get_core()->get_property(device_name, ov::device::capabilities);
+        auto support_model = std::find(capability.begin(), capability.end(), (model_precision));
+        bool is_support_fp16 =
+            model_precision == "FP32" && std::find(capability.begin(), capability.end(), ("FP16")) != capability.end();
+        return support_model != capability.end() || is_support_fp16;
+    };
 
-    for (auto& item : meta_devices) {
-        if (item.device_name.find("CPU") == 0) {
-            CPU.push_back(item);
-            continue;
-        }
-        if (item.device_name.find("MYRIAD") == 0) {
-            MYRIAD.push_back(item);
-            continue;
-        }
-        if (item.device_name.find("VPUX") == 0) {
-            VPUX.push_back(item);
+    for (auto&& device_info : meta_devices) {
+        if (device_info.device_priority > 0)
+            is_default_list = false;
```

Comment:
we don't need to judge this flag for each meta_devices, once is enough?

Reply:
btw, why do we need to know if is_default_list? default or not default, we already know from device_priority, isn't it?

Reply:
Let's keep this logic as discussed in the meeting.